### PR TITLE
Add support for playing a per-ruleset sound when switching rulesets

### PR DESCRIPTION
**Pull request by nekodex**
I looked at doing this at a `Ruleset` level ala `CreateIcon()`, but instantiating a `Sample` there seemed... not ideal? So I went with this approach instead.

Samples for the base rulesets are added in ppy/osu-resources#122 which will be played when the ruleset is selected (via the Toolbar).

### CODE REVIEW

File: osu.Game/Overlays/Toolbar/ToolbarRulesetSelector.cs [hunk 1]
```diff
@@ -54,6 +59,9 @@ private void load()
                     }
                 }
             });
+
+            foreach (var ruleset in Rulesets.AvailableRulesets)
+                selectionSamples[ruleset.ShortName] = audio.Samples.Get($"UI/ruleset-select-{ruleset.ShortName}");
```
Comment (peppy) [inline osu.Game/Overlays/Toolbar/ToolbarRulesetSelector.cs:64]: You mentioned that custom rulesets can also use this, but they won't be able to with this logic as it stands. The resource would need to be read from their storage, not the game's one.

Not sure if this is an issue just yet though. Might be something we add support for in the future.

Comment (nekodex) [reply osu.Game/Overlays/Toolbar/ToolbarRulesetSelector.cs:64]: Oh, I didn't realise that custom ruleset resources were independent... bad assumption on my behalf about how resources worked then, oops.

Comment (peppy) [reply osu.Game/Overlays/Toolbar/ToolbarRulesetSelector.cs:64]: Something similar to `CreateIcon()` is likely what we want, but I would probably recommend not doing that just yet due to concerns mentioned in #11240.

**Review by bdach — COMMENTED** [review]
Seems alright to get in as a first step. Can work on a ruleset API once we've figured out how to do the icons, too.

Not approving because conflicts.

**Review by peppy — APPROVED** [review]

**Pull request merged**